tests/commands/test_store_commands.py

```diff
@@ -533,3 +544,437 @@ def test_revisions_errors_multiple(caplog, store_mock):
+
+    tstamp_with_timezone = parser.parse('2020-07-03T20:30:40Z')
+    channel_map = [
+        Release(revision=5, channel='latest/beta', expires_at=None),
+        Release(revision=12, channel='latest/beta/mybranch', expires_at=tstamp_with_timezone),
+    ]
+    channels = _build_channels()
+    channels.append(
+        Channel(
+            name='latest/beta/mybranch', fallback='latest/beta',
+            track='latest', risk='beta', branch='mybranch'))
+    revisions = [
+        _build_revision(revno=5, version='5.1'),
+        _build_revision(revno=12, version='ver.12'),
+    ]
+    store_mock.list_releases.return_value = (channel_map, channels, revisions)
+
+    args = Namespace(name='testcharm')
+    StatusCommand('group').run(args)
+
+    assert store_mock.mock_calls == [
+        call.list_releases('testcharm'),
+    ]
+
+    expected = [
+        "Track    Channel        Version    Revision    Expires at",
+        "latest   stable         -          -",
+        "         candidate      -          -",
+        "         beta           5.1        5",
+        "         edge           ↑          ↑",
+        "         beta/mybranch  ver.12     12          2020-07-03T20:30:40+00:00",
+    ]
+    assert expected == [rec.message for rec in caplog.records]
+
+
+def test_status_with_multiple_branches(caplog, store_mock):
+    """Support having multiple branches."""
+    caplog.set_level(logging.INFO, logger="charmcraft.commands")
+
+    tstamp_with_timezone = parser.parse('2020-07-03T20:30:40Z')
+    channel_map = [
+        Release(revision=5, channel='latest/beta', expires_at=None),
+        Release(revision=12, channel='latest/beta/branch-1', expires_at=tstamp_with_timezone),
+        Release(revision=15, channel='latest/beta/branch-2', expires_at=tstamp_with_timezone),
+    ]
+    channels = _build_channels()
+    channels.extend([
+        Channel(
+            name='latest/beta/branch-1', fallback='latest/beta',
+            track='latest', risk='beta', branch='branch-1'),
+        Channel(
+            name='latest/beta/branch-2', fallback='latest/beta',
+            track='latest', risk='beta', branch='branch-2'),
+    ])
+    revisions = [
+        _build_revision(revno=5, version='5.1'),
+        _build_revision(revno=12, version='ver.12'),
+        _build_revision(revno=15, version='15.0.0'),
+    ]
+    store_mock.list_releases.return_value = (channel_map, channels, revisions)
+
+    args = Namespace(name='testcharm')
+    StatusCommand('group').run(args)
+
+    assert store_mock.mock_calls == [
+        call.list_releases('testcharm'),
+    ]
+
+    expected = [
+        "Track    Channel        Version    Revision    Expires at",
+        "latest   stable         -          -",
+        "         candidate      -          -",
+        "         beta           5.1        5",
+        "         edge           ↑          ↑",
+        "         beta/branch-1  ver.12     12          2020-07-03T20:30:40+00:00",
```

Comment:
This is how snapcraft does it. IIRC it's because if you put it after beta, it will break edge's `↑`.

In any case this may be revisited by UX people (even if we will diverge from snapcraft!)